packages/go_router/lib/src/builder.dart

```diff
@@ -163,34 +164,69 @@ class RouteBuilder {
 
       keyToPages.putIfAbsent(goRouteNavKey, () => <Page<dynamic>>[]).add(page);
 
-      _buildRecursive(context, matchList, startIndex + 1, pop, routerNeglect,
-          keyToPages, newParams, navigatorKey);
-    } else if (route is ShellRoute) {
+      return _buildRecursive(context, matchList, startIndex + 1, pop,
+              routerNeglect, keyToPages, newParams, navigatorKey) ??
+          state;
+    } else if (route is ShellRouteBase) {
+      if (startIndex + 1 >= matchList.matches.length) {
+        throw _RouteBuilderError('Shell routes must always have child routes');
```

Comment:
this should be an assert since this is technically impossible as the shellroutebase doesn't contribute to path during match, and if this happens, it is probably and internal bug in go_router.

Reply:
👍 Agreed.